lib/java/mapping/src/main/java/co/airy/mapping/ContentMapper.java

```diff
@@ -35,12 +46,54 @@ public ContentMapper(List<SourceMapper> sourceMappers, OutboundMapper outboundMa
             throw new Exception("can not map message source " + message.getSource());
         }
 
-        return sourceMapper.render(message.getContent());
+        final List<Content> contentList = sourceMapper.render(message.getContent());
+
+        if (storageHost == null) {
+            return contentList;
+        }
+
+        /*
```

Comment:
Usually not a fan of long methods, but this loop cannot be moved without forwarding the metadata and message, which is not much cleaner.

Reply:
It's not so bad :)